app/models/manageiq/providers/redhat/infra_manager/provision/disk.rb

```diff
@@ -0,0 +1,78 @@
+module ManageIQ::Providers::Redhat::InfraManager::Provision::Disk
+  def configure_disks
+    configure_dialog_disks
+    requested_disks = options[:disks_add]
+    if requested_disks.nil?
+      _log.info "Disks settings will be inherited from the template."
+      signal :customize_guest
+    else
+      add_disks(requested_disks)
+      signal :poll_add_disks_complete
```

Comment:
No, this does not need to be consistent across providers.  VMware supports adding disks which happens when the initial config spec is built.  Here it is happening as a post-creations step.  If we add support for another provider we could possible refactor this logic then.

@masayag I have two suggests:

1. Moving this method to the `state_machine.rb` file to keep the `signal` logic together.
2. Mark most of the methods in this file as `private`.